MdeModulePkg/Core/Pei/Memory/MemoryServices.c

```diff
@@ -650,6 +658,15 @@ PeiAllocatePages (
     FreeMemoryBottom = &(Hob.HandoffInformationTable->EfiFreeMemoryBottom);
   }
 
+  // MU_CHANGE START
+  // Check to make sure we aren't allocating memory in runtime buckets
+  SyncMemoryBuckets ();
+  if (CheckIfInRuntimeBoundary (*FreeMemoryTop)) {

```

Comment:
This check is to make sure pei memory allocations that aren't included in the pei memory buckets aren't being allocated within the predefined memory bucket space.  The only time we might care about the pages being allocated is if we're allocating memory before the memory bucket range that might run into it.  This case is impossible because the memory allocation process in PEI is contiguous, and the memory buckets are allocated on the first runtime memory allocation.  So if the starting address is within the range of the pei memory buckets we just need to move it forward outside of that range and we don't have to worry about it.